tools/common/Driver.cs

```diff
@@ -235,54 +272,110 @@ public static void Log (int min_verbosity, string format, params object [] args)
 
 		public const bool IsXAMCORE_4_0 = false;
 
-#if MONOMAC
-#pragma warning disable 0414
-		static string userTargetFramework = TargetFramework.Default.ToString ();
-#pragma warning restore 0414
-#endif
-
-		static TargetFramework? targetFramework;
-		public static bool HasTargetFramework {
-			get { return targetFramework.HasValue; }
-		}
+		static TargetFramework targetFramework;
 
 		public static TargetFramework TargetFramework {
-			get { return targetFramework.Value; }
+			get { return targetFramework; }
 			set { targetFramework = value; }
 		}
 
-		static void SetTargetFramework (string fx)
+		// We need to delay validating the target framework until we've parsed all the command line arguments,
+		// so first store it here, and then we call ValidateTargetFramework when we're done parsing the command
+		// line arguments.
+		static string target_framework;
+		static void SetTargetFramework (string value)
 		{
-#if MONOMAC
-			userTargetFramework = fx;
-#endif
+			target_framework = value;
+		}
+
+		static void ValidateTargetFramework ()
+		{
+			if (string.IsNullOrEmpty (target_framework))
+				throw ErrorHelper.CreateError (86, Errors.MX0086 /* A target framework (--target-framework) must be specified */);
 
+			var fx = target_framework;
 			switch (fx.Trim ().ToLowerInvariant ()) {
-#if MONOMAC
 			case "xammac":
 			case "mobile":
 			case "xamarin.mac":
-				targetFramework = TargetFramework.Xamarin_Mac_2_0;
-				break;
-#endif
+				targetFramework = TargetFramework.Xamarin_Mac_2_0_Mobile;
+				ErrorHelper.Warning (90, Errors.MX0090, /* The target framework '{0}' is deprecated. Use '{1}' instead. */ fx, targetFramework);
+				return;
 			default:
 				TargetFramework parsedFramework;
-				if (!Xamarin.Utils.TargetFramework.TryParse (fx, out parsedFramework))
+				if (!TargetFramework.TryParse (fx, out parsedFramework))
 					throw ErrorHelper.CreateError (68, Errors.MX0068, fx);
-#if MONOMAC
-				if (parsedFramework == TargetFramework.Net_3_0 || parsedFramework == TargetFramework.Net_3_5)
-					parsedFramework = TargetFramework.Net_2_0;
-#endif
 
 				targetFramework = parsedFramework;
 
 				break;
 			}
 
-#if MTOUCH
-			if (Array.IndexOf (TargetFramework.ValidFrameworks, targetFramework.Value) == -1)
-				throw ErrorHelper.CreateError (70, Errors.MT0070, targetFramework.Value, string.Join (" ", TargetFramework.ValidFrameworks.Select ((v) => v.ToString ()).ToArray ()));
+			bool show_0090 = false;
+#if MONOMAC
+			if (!TargetFramework.IsValidFramework (targetFramework)) {
+				// For historic reasons this is messy.
+				// If the TargetFramework we got isn't any of the one we accept, we have to do some fudging.
+				bool force45From40UnifiedSystemFull = false;
+
+				if (references.Any ((v) => Path.GetFileName (v) == "XamMac.dll"))
```

Comment:
there should not be any usage  of `XamMac.dll` *classic/32bits) anymore

Reply:
Exactly, the error we're showing is:

> Projects using the Classic API are not supported anymore. Please migrate the project to the Unified API.

Reply:
make sense - localization/using id does not help :| maybe a comment ?